app/services/product_price_handler.rb

```diff
@@ -0,0 +1,30 @@
+class ProductPriceHandler
+  include ProductsHelper
+
+  def initialize(product, flash, product_price, existing_product_price)
+    @product = product
+    @flash = flash
+    @product_price = product_price
+    @existing_product_price = existing_product_price
+  end
+
+  def call
+    return unless price_update
+
+    added_existing_prices_flash(
+      product_price.quantity, product.name, product_price.buy_price.format, product_price.sell_price.format
```

Comment:
flash messages should be on controllers level. You could do some presenter or whatever, but services only for business logic, and that should return true/false, and some additional data. But flash shouldn't be  here